src/node.cc

```diff
@@ -2502,18 +2506,44 @@ struct node_module* get_linked_module(const char* name) {
 // cache that's a plain C list or hash table that's shared across contexts?
 static void DLOpen(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
+  bool is_dlopen_error = false;
   uv_lib_t lib;
 
   CHECK_EQ(modpending, nullptr);
 
-  if (args.Length() != 2) {
-    env->ThrowError("process.dlopen takes exactly 2 arguments.");
+  if (args.Length() < 2) {
+    env->ThrowError("process.dlopen needs at least 2 arguments.");
     return;
   }
 
   Local<Object> module = args[0]->ToObject(env->isolate());  // Cast
   node::Utf8Value filename(env->isolate(), args[1]);  // Cast
-  const bool is_dlopen_error = uv_dlopen(*filename, &lib);
+
+#ifdef __POSIX__
+  // If no flag is passed, fallback to default
+  if (args.Length() < 3 || args[2]->IsUndefined() ||
+      args[2]->IsNull() || args[2]->IntegerValue() == 0) {
```

Comment:
`IntegerValue()` without arguments is deprecated, so it would be good if you could either use the overload that returns a `Maybe<int64_t>` and pass along the error if one occurs, or alternatively throw a type error if `args[2]->IsInt32()` fails and then use `args[2].As<Int32>()->Value()` to get the flags